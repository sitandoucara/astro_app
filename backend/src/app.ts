import express from "express";
import dotenv from "dotenv";
import cors from "cors";
import chartRoutes from "./routes/chartRoutes";
import timezoneRoutes from "./routes/timezoneRoutes";

dotenv.config();

const app = express();
app.use(cors());
app.use(express.json());

app.use("/api/chart", chartRoutes);
app.use("/api/timezone", timezoneRoutes);

const PORT = process.env.PORT || 4000;
app.listen(PORT, () => {
  console.log(`Server running on port ${PORT}`);
});
