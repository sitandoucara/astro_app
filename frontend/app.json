{
  "expo": {
    "name": "my-expo-app",
    "slug": "my-expo-app",
    "version": "1.0.9",
    "web": {
      "bundler": "metro",
      "favicon": "./assets/favicon.png"
    },
    "experiments": {
      "tsconfigPaths": true
    },
    "plugins": ["expo-font"],
    "orientation": "portrait",
    "icon": "./assets/icon.png",
    "userInterfaceStyle": "light",
    "splash": {
      "image": "./assets/splash.png",
      "resizeMode": "contain",
      "backgroundColor": "#ffffff"
    },
    "assetBundlePatterns": ["**/*"],
    "ios": {
      "supportsTablet": false,
      "bundleIdentifier": "com.sigraph.astromood",
      "infoPlist": {
        "ITSAppUsesNonExemptEncryption": false
      }
    },
    "android": {
      "adaptiveIcon": {
        "foregroundImage": "./assets/adaptive-icon.png",
        "backgroundColor": "#ffffff"
      }
    }
  }
}
