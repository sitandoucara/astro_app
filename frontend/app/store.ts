import AsyncStorage from '@react-native-async-storage/async-storage';
import { configureStore } from '@reduxjs/toolkit';
import { persistStore, persistReducer } from 'redux-persist';

import authReducer from '../features/auth/AuthSlice';
import languageReducer from '../shared/language/languageSlice';
import themeReducer from '../shared/theme/themeSlice';

const themePersistConfig = {
  key: 'theme',
  storage: AsyncStorage,
};

const languagePersistConfig = {
  key: 'language',
  storage: AsyncStorage,
};

const persistedThemeReducer = persistReducer(themePersistConfig, themeReducer);
const persistedLanguageReducer = persistReducer(languagePersistConfig, languageReducer);

export const store = configureStore({
  reducer: {
    auth: authReducer,
    theme: persistedThemeReducer,
    language: persistedLanguageReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: {
        ignoredActions: ['persist/PERSIST', 'persist/REHYDRATE'],
      },
    }),
});

export const persistor = persistStore(store);

export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
