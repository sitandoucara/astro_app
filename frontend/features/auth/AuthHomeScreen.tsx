import { useNavigation } from '@react-navigation/native';
import { NativeStackNavigationProp } from '@react-navigation/native-stack';
import { RootStackParamList } from 'navigation/types';
import { View, Text, TouchableOpacity, Image } from 'react-native';
import Animated, { FadeInUp } from 'react-native-reanimated';
import { useAppSelector } from 'shared/hooks';

export default function AuthHomeScreen() {
  const navigation = useNavigation<NativeStackNavigationProp<RootStackParamList>>();

  const isDarkMode = useAppSelector((state) => state.theme.isDarkMode);

  const textPrimary = isDarkMode ? 'text-light-text1' : 'text-[#F6D5C1]';
  const textSecondary = isDarkMode ? 'text-[#7B635A]' : 'text-[#ffffff]';
  const textButton1 = isDarkMode ? 'text-[#F2EAE0]' : 'text-[#281109]';
  const textButton2 = isDarkMode ? 'text-[#F2EAE0]' : 'text-[#281109]';
  const bgButton = isDarkMode ? 'bg-[#281109]' : 'bg-[#F2EAE0]';
  const bgButton2 = isDarkMode ? 'bg-[#281109]' : 'bg-[#F6D5C1]';
  const border = isDarkMode ? 'border-[#281109]' : 'border-[#F2EAE0]';
  const backgroundColor = isDarkMode ? '#F2EAE0' : '#281109';

  const logoSource = isDarkMode
    ? require('../../assets/logo_light.png')
    : require('../../assets/logo_dark.png');

  return (
    <View className="flex-1 p-16" style={{ backgroundColor }}>
      <View className="mt-10 flex-1 justify-between">
        <Animated.View entering={FadeInUp.duration(1000)} className="mt-5 items-center">
          <Image source={logoSource} style={{ width: 280, height: 280 }} resizeMode="contain" />
        </Animated.View>

        <View className="mb-5 mt-10 ">
          <View className="mb-8">
            <Text className={`text-aref mb-4 text-center text-4xl font-medium ${textPrimary}`}>
              Welcome
            </Text>
            <Text
              className={`text-aref text-center text-sm text-[#7B635A] ${textSecondary}`}
              style={{ lineHeight: 20 }}>
              You are on the path to discovering yourself. LunaFlame will help you live in harmony
              with the Universe.
            </Text>
          </View>

          {/* registration */}
          <Animated.View entering={FadeInUp.delay(300).duration(1000)}>
            <View className="w-full flex-row items-center justify-center gap-2">
              <View className="flex-row items-center" style={{ gap: 8 }}>
                <View className={`h-3 w-3 rounded-full ${bgButton}  `} />
                <View className={`h-4 w-4 rounded-full   ${bgButton}  `} />
              </View>
              <View className={`rounded-full border-2 ${border}  p-2 `}>
                <TouchableOpacity
                  onPress={() => navigation.navigate('SignUp')}
                  activeOpacity={0.8}
                  className={`shadow-opacity-30  elevation-1 rounded-full ${bgButton} px-16  py-3 shadow-md shadow-light-text2`}>
                  <Text
                    className={`text-aref text-center text-xl font-bold tracking-wide   ${textButton1}`}>
                    Create Account
                  </Text>
                </TouchableOpacity>
              </View>
              <View className="flex-row items-center" style={{ gap: 8 }}>
                <View className={`h-4 w-4 rounded-full  ${bgButton}  `} />
                <View className={`h-3 w-3 rounded-full   ${bgButton}  `} />
              </View>
            </View>

            {/*connection*/}
            <View className="mt-4 w-full flex-row items-center justify-center gap-2">
              <View className="flex-row items-center" style={{ gap: 8 }}>
                <View className={`h-1 w-8 rounded-full  ${bgButton2}  `} />
                <View className={`h-4 w-4 rounded-full   ${bgButton2}  `} />
              </View>
              <View className={`rounded-full border-2 ${border}   p-2 `}>
                <TouchableOpacity
                  onPress={() => navigation.navigate('SignIn')}
                  activeOpacity={0.8}
                  className={`shadow-opacity-30  elevation-1 rounded-full ${bgButton2} px-8  py-3 shadow-lg shadow-[#281109]`}>
                  <Text
                    className={`text-aref text-center text-base font-bold tracking-wide  ${textButton2} `}>
                    Already have an account
                  </Text>
                </TouchableOpacity>
              </View>
              <View className="flex-row items-center" style={{ gap: 8 }}>
                <View className={`h-4 w-4 rounded-full  ${bgButton2}  `} />
                <View className={`h-1 w-8 rounded-full   ${bgButton2}  `} />
              </View>
            </View>
          </Animated.View>
        </View>
      </View>
    </View>
  );
}
