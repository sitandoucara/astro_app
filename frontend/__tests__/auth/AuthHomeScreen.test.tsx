import { NavigationContainer } from '@react-navigation/native';
import { render, fireEvent } from '@testing-library/react-native';
import { store } from 'app/store';
import AuthHomeScreen from 'features/auth/AuthHomeScreen';
import React from 'react';
import { Provider } from 'react-redux';

// Mock navigation
const mockNavigate = jest.fn();
jest.mock('@react-navigation/native', () => {
  const actualNav = jest.requireActual('@react-navigation/native');
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: mockNavigate,
    }),
  };
});

// Render helper with Redux and Navigation
const renderWithProviders = (component: React.ReactElement) => {
  return render(
    <Provider store={store}>
      <NavigationContainer>{component}</NavigationContainer>
    </Provider>
  );
};

describe('AuthHomeScreen Tests', () => {
  it('A - AuthHomeScreen renders correctly (snapshot)', () => {
    const { toJSON } = renderWithProviders(<AuthHomeScreen />);
    expect(toJSON()).toMatchSnapshot();
  });

  it("B - 'Create Account' button navigates to SignUp", () => {
    const { getByText } = renderWithProviders(<AuthHomeScreen />);
    fireEvent.press(getByText('Create Account'));
    expect(mockNavigate).toHaveBeenCalledWith('SignUp');
  });

  it("C - 'Already have an account' button navigates to SignIn", () => {
    const { getByText } = renderWithProviders(<AuthHomeScreen />);
    fireEvent.press(getByText('Already have an account'));
    expect(mockNavigate).toHaveBeenCalledWith('SignIn');
  });
});
