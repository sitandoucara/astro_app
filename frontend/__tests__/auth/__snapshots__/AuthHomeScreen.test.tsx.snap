// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthHomeScreen Tests A - AuthHomeScreen renders correctly (snapshot) 1`] = `
<View
  className="flex-1 p-16"
  style={
    {
      "backgroundColor": "#281109",
    }
  }
>
  <View
    className="mt-10 flex-1 justify-between"
  >
    <View
      className="mt-5 items-center"
      collapsable={false}
      entering={
        FadeInUp {
          "build": [Function],
          "durationV": 1000,
          "randomizeDelay": false,
          "reduceMotionV": "system",
        }
      }
      jestAnimatedStyle={
        {
          "value": {},
        }
      }
    >
      <Image
        resizeMode="contain"
        source=""
        style={
          {
            "height": 280,
            "width": 280,
          }
        }
      />
    </View>
    <View
      className="mb-5 mt-10 "
    >
      <View
        className="mb-8"
      >
        <Text
          className="text-aref mb-4 text-center text-4xl font-medium text-[#F6D5C1]"
        >
          Welcome
        </Text>
        <Text
          className="text-aref text-center text-sm text-[#7B635A] text-[#ffffff]"
          style={
            {
              "lineHeight": 20,
            }
          }
        >
          You are on the path to discovering yourself. LunaFlame will help you live in harmony with the Universe.
        </Text>
      </View>
      <View
        collapsable={false}
        entering={
          FadeInUp {
            "build": [Function],
            "delayV": 300,
            "durationV": 1000,
            "randomizeDelay": false,
            "reduceMotionV": "system",
          }
        }
        jestAnimatedStyle={
          {
            "value": {},
          }
        }
      >
        <View
          className="w-full flex-row items-center justify-center gap-2"
        >
          <View
            className="flex-row items-center"
            style={
              {
                "gap": 8,
              }
            }
          >
            <View
              className="h-3 w-3 rounded-full bg-[#F2EAE0]  "
            />
            <View
              className="h-4 w-4 rounded-full   bg-[#F2EAE0]  "
            />
          </View>
          <View
            className="rounded-full border-2 border-[#F2EAE0]  p-2 "
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-aref text-center text-xl font-bold tracking-wide   text-[#281109]"
              >
                Create Account
              </Text>
            </View>
          </View>
          <View
            className="flex-row items-center"
            style={
              {
                "gap": 8,
              }
            }
          >
            <View
              className="h-4 w-4 rounded-full  bg-[#F2EAE0]  "
            />
            <View
              className="h-3 w-3 rounded-full   bg-[#F2EAE0]  "
            />
          </View>
        </View>
        <View
          className="mt-4 w-full flex-row items-center justify-center gap-2"
        >
          <View
            className="flex-row items-center"
            style={
              {
                "gap": 8,
              }
            }
          >
            <View
              className="h-1 w-8 rounded-full  bg-[#F6D5C1]  "
            />
            <View
              className="h-4 w-4 rounded-full   bg-[#F6D5C1]  "
            />
          </View>
          <View
            className="rounded-full border-2 border-[#F2EAE0]   p-2 "
          >
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-aref text-center text-base font-bold tracking-wide  text-[#281109] "
              >
                Already have an account
              </Text>
            </View>
          </View>
          <View
            className="flex-row items-center"
            style={
              {
                "gap": 8,
              }
            }
          >
            <View
              className="h-4 w-4 rounded-full  bg-[#F6D5C1]  "
            />
            <View
              className="h-1 w-8 rounded-full   bg-[#F6D5C1]  "
            />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
